cmd/launcher/secure_enclave_darwin.go

```diff
@@ -0,0 +1,182 @@
+//go:build darwin
+// +build darwin
+
+package main
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/kolide/krypto/pkg/challenge"
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/kolide/krypto/pkg/secureenclave"
+	"github.com/kolide/launcher/ee/agent/certs"
+	"github.com/kolide/launcher/ee/secureenclavesigner"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+const secureEnclaveTimestampValiditySeconds = 150
+
+var serverPubKeys = make(map[string]*ecdsa.PublicKey)
+
+// runSecureEnclave performs either a create-key or sign operation using the secure enclave.
+// It's available as a separate command because launcher runs aa root by default and since it's
+// not in a user security context, it can't use the secure enclave directly. However, this command
+// can be run in the user context using launchctl. To perform an operation, root launcher needs to
+// include a challenge signed by a known server. See ee/secureenclavesigner for command data
+// structure.
+func runSecureEnclave(args []string) error {
+	if len(args) < 2 {
+		return errors.New("not enough arguments, expect create_key <request> or sign <sign_request>")
+	}
+
+	if err := loadServerKeys(); err != nil {
+		return fmt.Errorf("loading server keys: %w", err)
+	}
+
+	if args[1] == "" {
+		return errors.New("missing request")
+	}
+
+	switch args[0] {
```

Comment:
Thinking about deletion/uninstall. 

As it stands today, the uninstall message is not signed. So how do we uninstall these keys? Do we need to?

If we need to delete them, we probably need a signed deletion message.

If we don't need to, we could do somewhere where launcher records a nonce in the launcherdb, and then we pass the nonce along to the sign command. (so that we can _find_ the correct key). (And by nonce, I probably mean the public key, we could stash it in the launcher db. Though that will be weird cross users. Probably still correct)